### PR TITLE
Added a new argument called key-properties

### LINKED ISSUE #5: Add attributes from Geojson to SVG
Hello,

Can we add some attributes from Geojson to SVG, with providing which properties or geometries in Geojson we need to add in SVG, with "geojson-to-svg-cli"?

### PR DESCRIPTION
**Pull request by nestoralonso**
Fixes #5

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -19,10 +19,11 @@ if (argv.help || argv.h) {
 Usage:
     geojson-to-svg [-p mercator]
 Options:
-    --projection  -p  Which projection to use. Default: mercator
-                      Refer to npmjs.com/projections for a list.
+	--projection  -p  Which projection to use. Default: mercator
```
Comment (derhuerst) [inline cli.js]: Please keep the indentation as before.

File: cli.js [hunk 2]
```diff
@@ -19,10 +19,11 @@ if (argv.help || argv.h) {
...
+	Refer to npmjs.com/projections for a list.
+	--key-properties  -k  comma separated list of properties to add to the svg as data attributes.
```
Comment (derhuerst) [inline cli.js]: Let's call it just `--properties`/`-p`.

File: cli.js [hunk 3]
```diff
@@ -65,7 +69,25 @@ process.stdin
...
-		const polylines = svgify(geojson, {projection: project})
+		const polylines = svgify(geojson, {
+			projection: project,
+			computeProps(feature) {
+				const {properties} = feature
+				const dataAttributes = Object.entries(properties).reduce((acc, entry) => {
+					const [key, value] = entry
+					if (!propertiesArray.includes(key)) return acc;
+
+					const dataKey = `data-${key.toLowerCase().replace('_', '-')}`
```
Comment (derhuerst) [inline cli.js]: Please keep the original key name.

File: cli.js [hunk 4]
```diff
@@ -65,7 +69,25 @@ process.stdin
...
+					acc[dataKey] = value
+
+					return acc
+				}, {})
+				return Object.assign({
```
Comment (derhuerst) [inline cli.js]: Use `dataAttributes.className = 'shape'; return dataAttributes` to make sure `className` always has `shape` as value.

**Comment by derhuerst** [comment]
Thanks for your contribution! I have a few comments.

File: cli.js (hunk 2 above)
Comment (nestoralonso) [reply cli.js]: But if I leave it that way the name will collide with --projection  -p , maybe change the short name to --properties/-k, what are your thoughts?

File: cli.js (hunk 3 above)
Comment (nestoralonso) [reply cli.js]: The name of the data attributes must comply with this spec:
Embedding custom non-visible data with the data-* attributes
A custom data attribute is an attribute in no namespace whose name starts with the string "data-", has at least one character after the hyphen, is XML-compatible, and contains no uppercase ASCII letters.
https://stackoverflow.com/questions/19533402/naming-rules-for-html5-custom-data-attributedata

File: cli.js (hunk 2 above)
Comment (derhuerst) [reply cli.js]: Good point. Let's then use `--key`/`-k`.

File: cli.js (hunk 3 above)
Comment (derhuerst) [reply cli.js]: Then let's do `key.toLowerCase()` only. My intention is to keep `geojson-to-svg` as least magic as possible.

File: cli.js [hunk 5]
```diff
@@ -19,10 +19,12 @@ if (argv.help || argv.h) {
 Usage:
     geojson-to-svg [-p mercator]
 Options:
-    --projection  -p  Which projection to use. Default: mercator
-                      Refer to npmjs.com/projections for a list.
+	--projection  -p	Which projection to use. Default: mercator
```
Comment (derhuerst) [inline cli.js]: The indentation is still different.

File: cli.js [hunk 6]
```diff
@@ -65,7 +70,24 @@ process.stdin
 		const geojson = JSON.parse(input)
 		const [west, south, east, north] = bbox(geojson)
 
-		const polylines = svgify(geojson, {projection: project})
+		const polylines = svgify(geojson, {
+			projection: project,
+			computeProps(feature) {
+				const {properties} = feature
+				const dataAttributes = Object.entries(properties).reduce((acc, entry) => {
```
Comment (derhuerst) [inline cli.js]: This breaks Node 6. Let's use a good old `for (const key of Object.keys(properties))` here.

File: cli.js (hunk 5 above)
Comment (nestoralonso) [reply cli.js]: Ok, I'll fixit

Comment (derhuerst) [reply cli.js]: Thanks! 💛

**Comment by derhuerst** [comment]
Almost done! Thanks for being patient.

File: cli.js (hunk 6 above)
Comment (nestoralonso) [reply cli.js]: Ok

File: cli.js (hunk 5 above)
Comment (nestoralonso) [reply cli.js]: Maybe VSCode is messing with that, I'll try another editor

**Comment by derhuerst** [comment]
Thanks again!

**Pull request merged**